src/main/scala/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.scala

```diff
@@ -340,6 +340,7 @@ class CollectDuplexSeqMetrics
 
       // Assign reads a random number between 0 and 1 inclusive based on their read name
       group.foreach { rec =>
+        if (rec.isConsensus) throw new IllegalArgumentException("Found consensus record. Expected UMI-grouped BAM")
```

Comment:
This validation could be raised much earlier and external to the `collect()` scope. Perhaps `.tapEach()` and make the exception raise on the `filteredIterator` in the tool itself? You could also wrap the filtered iterator in a peekable iterator, perform the validation on the first record if it exists, and then send the peekable iterator into `.collect()`.